src/protocols/output_management.rs

```diff
@@ -0,0 +1,791 @@
+                        }
+                    }
+                }
+                (Some(_), None) => {
+                    changed = true;
+                    for (_, client) in &state.clients {
+                        if let Some((head, _)) = client.heads.get(name) {
+                            head.enabled(0);
+                        }
+                    }
+                }
+                (None, Some(new_index)) => {
+                    changed = true;
+                    for (_, client) in &state.clients {
+                        if let Some((head, _)) = client.heads.get(name) {
+                            head.enabled(1);
+                            if old.modes.len() <= new_index || new.modes.len() <= new_index {
+                                error!("notify_changes: out of bound mode");
+                            } else if old.modes[new_index] == new.modes[new_index] {
+                                for (_, client) in &state.clients {
+                                    if let Some((head, modes)) = client.heads.get(name) {
+                                        head.current_mode(&modes[new_index]);
+                                    }
+                                }
+                            } else {
+                                // The mod has not been registerd with the head :/
+                            }
+                        }
+                    }
+                }
+                (None, None) => {}
+            }
+            match (old.logical, new.logical) {
+                (Some(old_logical), Some(new_logical)) => {
+                    if old_logical != new_logical {
+                        changed = true;
+                        for (_, client) in &state.clients {
+                            if let Some((head, _)) = client.heads.get(name) {
+                                if old_logical.x != new_logical.x || old_logical.y != old_logical.x
+                                {
+                                    head.position(new_logical.x, new_logical.y);
+                                }
+                                if old_logical.scale != new_logical.scale {
+                                    head.scale(new_logical.scale);
+                                }
+                                if old_logical.transform != new_logical.transform {
+                                    head.transform(
+                                        ipc_transform_to_smithay(new_logical.transform).into(),
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+                (None, Some(new_logical)) => {
+                    changed = true;
+                    for (_, client) in &state.clients {
+                        if let Some((head, _)) = client.heads.get(name) {
+                            head.enabled(0);
```

Comment:
First of all this should probably be 1, but also is this really needed here? It's already sent in the modes check above.